homeassistant/components/google_assistant/trait.py

```diff
@@ -1262,16 +1307,19 @@ def query_attributes(self):
                 mode_settings["sound mode"] = attrs.get(media_player.ATTR_SOUND_MODE)
         elif self.state.domain == input_select.DOMAIN:
             mode_settings["option"] = self.state.state
+        elif self.state.domain == humidifier.DOMAIN:
+            if humidifier.ATTR_MODE in attrs:
+                mode_settings["mode"] = attrs.get(humidifier.ATTR_MODE)
 
         if mode_settings:
-            response["on"] = self.state.state != STATE_OFF
+            response["on"] = self.state.state not in (STATE_OFF, STATE_UNKNOWN)
             response["online"] = True
```

Comment:
We should drop "online", it's already set to true.

Reply:
Done. Also removed for `FanSpeedTrait` if you don't mind.